coredb-operator/src/fixtures.rs

```diff
@@ -0,0 +1,150 @@
+
+            let response = serde_json::to_vec(&coredb.finalized()).unwrap(); // respond as the apiserver would have
+            send.send_response(Response::builder().body(Body::from(response)).unwrap());
+        })
+    }
+
+    pub fn handle_event_publish(self) -> JoinHandle<()> {
+        let handle = self.0;
+        tokio::spawn(async move {
+            pin_mut!(handle);
+            let (request, send) = handle.next_request().await.expect("service not called");
+            assert_eq!(request.method(), http::Method::POST);
+            assert_eq!(
+                request.uri().to_string(),
+                format!("/apis/events.k8s.io/v1/namespaces/testns/events?")
+            );
+            // pass the event straight through
+            send.send_response(Response::builder().body(request.into_body()).unwrap());
+        })
+    }
+
+    pub fn handle_coredb_patch(self, coredb_: &CoreDB) -> JoinHandle<()> {
+        let handle = self.0;
+        let coredb = coredb_.clone();
+        tokio::spawn(async move {
+            pin_mut!(handle);
+            // expecting to get a PATCH request to update CoreDB resource
+            let (request, send) = handle
+                .next_request()
+                .await
+                .expect("Kube API called to PATCH CoreDB");
+            assert_eq!(request.method(), http::Method::PATCH);
+            assert_eq!(
+                request.uri().to_string(),
+                format!(
+                    "/apis/kube.rs/v1/namespaces/testns/coredbs/{}/status?&force=true&fieldManager=cntrlr",
+                    coredb.name_any()
+                )
+            );
+            let req_body = to_bytes(request.into_body()).await.unwrap();
+            let json: serde_json::Value =
+                serde_json::from_slice(&req_body).expect("patch_status object is json");
+            let status_json = json.get("status").expect("status object").clone();
+            let status: CoreDBStatus = serde_json::from_value(status_json).expect("contains valid status");
+            assert_eq!(
+                status.running, true,
+                "CoreDB::test says the status isn't running, but it was expected to be running."
+            );
+
+            let response = serde_json::to_vec(&coredb.with_status(status)).unwrap();
+            // pass through coredb "patch accepted"
+            send.send_response(Response::builder().body(Body::from(response)).unwrap());
+
+            // After the PATCH to CoreDB, we expect a PATCH to StatefulSet
+            let (request, send) = handle
+                .next_request()
+                .await
+                .expect("Kube API called to PATCH StatefulSet");
+            assert_eq!(request.method(), http::Method::PATCH);
+            assert_eq!(
+                request.uri().to_string(),
+                format!("/apis/apps/v1/namespaces/testns/statefulsets/testdb?&force=true&fieldManager=cntrlr")
+            );
```

Comment:
In this test, we first get the PATCH to the CoreDB object, then wait for the next request, and check that it's to PATCH the corresponding StatefulSet.